[WebAuthn] Add SPI to set name of platform authenticator credential

pascoej opened the pull request:
#### e84ec05033e0d2d24ead851f62d6ac431610230a
<pre>
[WebAuthn] Add SPI to set name of platform authenticator credential
<a href="https://bugs.webkit.org/show_bug.cgi?id=242403">https://bugs.webkit.org/show_bug.cgi?id=242403</a>

Reviewed by Brent Fulgham.

This SPI is needed to modify credentials according to internal needs.

* Source/WebCore/Modules/webauthn/fido/FidoConstants.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
(+[_WKWebAuthenticationPanel setDisplayNameForLocalCredentialWithGroupAndID:credential:displayName:]):
(+[_WKWebAuthenticationPanel setNameForLocalCredentialWithGroupAndID:credential:name:]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/_WKWebAuthenticationPanel.mm:
(TestWebKitAPI::TEST):

Canonical link: <a href="https://commits.webkit.org/252292@main">https://commits.webkit.org/252292@main</a>
</pre>

brentfulgham commented on Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
```diff
@@ -428,6 +428,70 @@ + (void)setDisplayNameForLocalCredentialWithGroupAndID:(NSString *)group credent
+        ASSERT_NOT_REACHED();
+        return;
+    }
+    NSDictionary *attributes = (__bridge NSDictionary *)attributesArrayRef;
```
The `attributesArrayRef` returned by `SecItemCopyMatching` has a +1 reference count. You need to adopt that into a RetainPtr<> so you can early return without leaking memory.

Maybe:
`auto attributes = adoptNS((__bridge NSDictionary *)attributesArrayRef);`

brentfulgham reviewed (approved):
Looks good, except for the leak. r=me with that change.

pascoej committed:
[WebAuthn] Add SPI to set name of platform authenticator credential
https://bugs.webkit.org/show_bug.cgi?id=242403

Reviewed by Brent Fulgham.

This SPI is needed to modify credentials according to internal needs.

* Source/WebCore/Modules/webauthn/fido/FidoConstants.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.h:
* Source/WebKit/UIProcess/API/Cocoa/_WKWebAuthenticationPanel.mm:
(+[_WKWebAuthenticationPanel setDisplayNameForLocalCredentialWithGroupAndID:credential:displayName:]):
(+[_WKWebAuthenticationPanel setNameForLocalCredentialWithGroupAndID:credential:name:]):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/_WKWebAuthenticationPanel.mm:
(TestWebKitAPI::TEST):

Canonical link: https://commits.webkit.org/252292@main
